Python/Product/PythonTools/visualstudio_py_util.py

```diff
@@ -134,7 +134,9 @@ def exec_module(module, global_variables):
 def read_bytes(conn, count):
     b = to_bytes('')
     while len(b) < count:
-        b += conn.recv(count - len(b))
+        received_data = conn.recv(count - len(b))
+        if received_data:
+            b += received_data
```

Comment:
For a TCP connection, AFAIK, you can get a zero-length read if the other side did a zero-length write in the middle of a stream, so it's not necessarily a reliable EOF marker. But it should still be an empty byte array in that case, and so it'd be safe to `+=` it. If this rather checks for `None`, I don't think that can happen in any case other than connection closing.